File: Count Good Numbers/Count Good Numbers.py
# Solution for Count Good Numbers in PY

class Solution:

    def __init__(self):
        self.M = 1_000_000_007
    
    def findPower(self, a: int, b: int) -> int:
        if b == 0:
            return 1
        half = self.findPower(a, b // 2)
        result = (half * half) % self.M
        if b % 2 == 1:
            result = (result * a) % self.M
        return result

    def countGoodNumbers(self, n: int) -> int:
        return self.findPower(5, (n + 1) // 2) * self.findPower(4, n // 2) % self.M

File: Count Good Numbers/Count Good Numbers.java
// Solution for Count Good Numbers in JAVA

class Solution {

    private static final int M = 1_000_000_007;
    private long findPower(long a, long b) {
        if (b == 0) return 1;
        long half = findPower(a, b / 2);
        long result = (half * half) % M;
        if (b % 2 == 1) {
            result = (result * a) % M;
        }
        return result;
    }

    public int countGoodNumbers(long n) {
        return (int)((findPower(5, (n + 1) / 2) * findPower(4, n / 2)) % M);
    }
}
